web/themes/custom/nysenate_theme/nysenate_theme.theme

```diff
@@ -3151,10 +3151,21 @@ function nysenate_theme_preprocess_page__node__microsite_page(&$variables) {
     }
   }
 
-  // Override the main content.
-  // Display the microsite node with altered contents.
-  $variables['page']['content']['system_main'] = \Drupal::entityTypeManager()->getViewBuilder('node')
-    ->view($node, 'full');
+  // Override the main content, if not in a Webform confirmation context.
+  $route_name = \Drupal::routeMatch()->getRouteName();
+  $in_webform_confirm_context = (
```

Comment:
@nathanielwoodland pls explain the logic behind resolving `$in_webform_confirm_context`

Reply:
@routinet, sure: there are two conditions that indicate if the current request is coming from a webform confirmation page context. The obvious one -- and I suspect the primary one, now that the underlying bug is fixed -- is `$route_name` == 'entity.node.webform.confirmation'`. However, if the webform is setup to render the confirmation message "inline" on the node route, then we can see that we're in a confirmation context due to the `webform/webform.confirmation` library being attached to the page. This second condition is more of a fallback, as I suspect most forms will use the default confirmation route. Happy to chat further about this.

Reply:
what are the chances of false positives here?  E.g., the library being attached outside of a confirmation?

Reply:
Good question. I would think very low. Webform is one of the most popular contrib modules, and there would have to be a bug in the code for it to inject this library into another Drupal context.